src/search.cpp

```diff
@@ -259,27 +259,23 @@ void MainThread::search() {
 
       // Find out minimum score and reset votes for moves which can be voted
       for (Thread* th: Threads)
-      {
           minScore = std::min(minScore, th->rootMoves[0].score);
-          votes[th->rootMoves[0].pv[0]] = 0;
-      }
 
       // Vote according to score and depth
-      auto square = [](int64_t x) { return x * x; };
       for (Thread* th : Threads)
-          votes[th->rootMoves[0].pv[0]] += 200 + (square(th->rootMoves[0].score - minScore + 1)
-                                                  * int64_t(th->completedDepth));
+      {
+          int64_t s = th->rootMoves[0].score - minScore + 1;
+          votes[th->rootMoves[0].pv[0]] += 200 + s * s * int(th->completedDepth);
+      }
 
       // Select best thread
-      int64_t bestVote = votes[this->rootMoves[0].pv[0]];
+      auto bestVote = votes[this->rootMoves[0].pv[0]];
```

Comment:
@mcostalba Just curious why you like auto here?  I generally dislike auto since then I can't tell what type is actually being used without having to search for some other part of the code to find out.  (I mind it a bit less if the type has a convoluted syntax like a function pointer or a container iterator so it looks cleaner.)

Reply:
Because bestVote is:

- temporary variable
- initialized by and assignment to votes[] and used to compare just with votes[]

So we really don't **HAVE** to care about it's real type, just that **it should be teh same type of vote[]**. Think of it as a kind of enum, we don't want to know MOVE_NONE and WHITE are both integer types (and of zero value), we just want to know the first is Move enum and the latter is Color enum.

I hope  I have been clear enough.

Reply:
Because bestVote is:

    temporary variable
    initialized by and assignment to votes[] and used to compare just with votes[]

So we really don't HAVE to care about it's real type, just that it should be teh same type of vote[]. Think of it as a kind of enum, we don't want to know MOVE_NONE and WHITE are both integer types (and of zero value), we just want to know the first is Move enum and the latter is Color enum.

I hope I have been clear enough.

Reply:
Ok thanks.  That makes sense.